### PR TITLE
Add ssl_user_name vhost parameter

### PR DESCRIPTION
**Pull request by bodgit**
Maps to SSLUserName directive.

### CODE REVIEW

File: manifests/vhost.pp [hunk 1]
```diff
@@ -1660,6 +1660,9 @@
 # @param ssl_proxyengine
 #   Specifies whether or not to use [SSLProxyEngine](https://httpd.apache.org/docs/current/mod/mod_ssl.html#sslproxyengine).
 #
+# @param ssl_require
+#   Sets the [SSLRequire](https://httpd.apache.org/docs/current/mod/mod_ssl.html#sslrequire) directive.
```
Comment (ekohl) [inline manifests/vhost.pp:1664]: This states:
> SSLRequire is deprecated and should in general be replaced by Require expr.

Is it really needed to add a deprecated option?

Comment (bodgit) [reply manifests/vhost.pp:1664]: If I can convince my users to use that, sure. Where/How do you set that, looking for the word "require" in a Puppet codebase brings up a few hits 😉

Comment (ekohl) [reply manifests/vhost.pp:1664]: I think they can be set via the `directories` parameter. It supports a lot of options and and one of them is `require`.

Comment (bodgit) [reply manifests/vhost.pp]: I've removed that parameter now.

**Comment by bodgit** [comment]
Updated to just add the `ssl_user_name` parameter given `SSLRequire` is a deprecated directive.

File: spec/defines/vhost_spec.rb [hunk 2]
```diff
@@ -85,6 +85,7 @@
               'ssl_proxyengine'             => true,
               'ssl_proxy_cipher_suite'      => 'HIGH',
               'ssl_proxy_protocol'          => 'TLSv1.2',
+              'ssl_user_name'               => 'SSL_CLIENT_S_DN_CN',
```
Comment (igalic) [inline spec/defines/vhost_spec.rb:88]: where is the verification of this spec addition?

**Review by igalic — COMMENTED** [review]
🙋🏻‍♀️

File: spec/defines/vhost_spec.rb (hunk 2 above)
Comment (bodgit) [reply spec/defines/vhost_spec.rb:88]: It looks like a lot of the SSL options aren't actually verified https://github.com/puppetlabs/puppetlabs-apache/blob/bd7942973532003ee00c8a3e32fc4399ef75ee28/spec/defines/vhost_spec.rb#L1041

I'll add something for this.

Comment (bodgit) [reply spec/defines/vhost_spec.rb:88]: Done.

**Review by david22swan — CHANGES_REQUESTED** [review]
Thank you for the added functionality, have just reviewed this and while it look's good overall to me I would like to see an acceptance test added to cover the functionality. Doesn't need to be complex, just an instance of it being set:
https://github.com/puppetlabs/puppetlabs-apache/blob/main/spec/acceptance/vhost_spec.rb
If you can do this I would feel happy merging.

**Comment by sanfrancrisko** [comment]
Thanks for the enhancement @bodgit - as @david22swan mentioned [here|https://github.com/puppetlabs/puppetlabs-apache/pull/2093#pullrequestreview-540773693], if we can get an acceptance test added, then we'd be happy to merge. Please let me know if we can be of any assistance in that regard.

**Comment by bodgit** [comment]
Acceptance test added, and corrected a small typo.

**Review by sanfrancrisko — APPROVED** [review]
Thanks for the enhancement @bodgit and the review help @ekohl , @igalic , @david22swan

**Pull request merged**